src/dist/replication/meta_server/meta_server_failure_detector.cpp

```diff
@@ -263,6 +263,10 @@ void meta_server_failure_detector::on_ping(const fd::beacon_msg& beacon, ::dsn::
     {
         ack.is_master = false;
         ack.primary_node = get_primary();
+        {
```

Comment:
R considers itself alive as long as an alive meta server exists, no matter it is a master or not. We discussed about this before. And this is mainly for the performance. Because we don't want the switching of meta server to influence the liveness of the replica server.